redis/src/test/scala/scalacache/redis/ShardedRedisCacheSpec.scala

```diff
@@ -0,0 +1,132 @@
+package scalacache.redis
+
+import org.scalatest.concurrent.{ Eventually, IntegrationPatience, ScalaFutures }
+import org.scalatest.time.{ Seconds, Span }
+import org.scalatest.{ BeforeAndAfter, FlatSpec, Matchers }
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class ShardedRedisCacheSpec
```

Comment:
Could you refactor this spec to share the actual tests with `RedisCacheSpec`? In other words, there should be a common trait that holds all the test code, and `RedisCacheSpec` and `ShardedRedisCacheSpec` should just take care of setting up and tearing down the cache according to their respective implementations.